src/borg/testsuite/repository.py

```diff
@@ -655,7 +698,7 @@ def make_auxiliary(repository):
 
 
 def do_commit(repository):
-    with repository:
+    with reopen(repository) as repository:
```

Comment:
Since do_commit was using a CM, I think it is best to reopen the repo? In the tests that call this, I ensure that it is not called within other CM's.

Reply:
Not sure I understand the question.

The misc. tests calling `do_commit` get a fresh `repository` each.

Also as the repo CM is opening/closing the repo, i think the commit has to happen while the repo context is entered (while the repo is opened).

Reply:
Each of the do_commit calls in the tests happen after the original repo CM has exited though, so wouldn't we be reopening the repo in these cases?

Reply:
yes, indeed.